src/components/MessagesList/MessagesGroup/MessagesSystemGroup.vue

```diff
@@ -0,0 +1,614 @@
+							: t('spreed', '{actor} demoted you and {user0} from moderators')
+					} else {
+						combinedMessage.message = actorIsAdministrator
+							? n('spreed',
+								'An administrator demoted you, {user0} and %n more participant from moderators',
+								'An administrator demoted you, {user0} and %n more participants from moderators', usersCounter - 2)
+							: n('spreed',
+								'{actor} demoted you, {user0} and %n more participant from moderators',
+								'{actor} demoted you, {user0} and %n more participants from moderators', usersCounter - 2)
+					}
+				} else {
+					if (usersCounter === 2) {
+						combinedMessage.message = actorIsAdministrator
+							? t('spreed', 'An administrator demoted {user0} and {user1} from moderators')
+							: t('spreed', '{actor} demoted {user0} and {user1} from moderators')
+					} else {
+						combinedMessage.message = actorIsAdministrator
+							? n('spreed',
+								'An administrator demoted {user0}, {user1} and %n more participant from moderators',
+								'An administrator demoted {user0}, {user1} and %n more participants from moderators', usersCounter - 2)
+							: n('spreed',
+								'{actor} demoted {user0}, {user1} and %n more participant from moderators',
+								'{actor} demoted {user0}, {user1} and %n more participants from moderators', usersCounter - 2)
+					}
+				}
+			}
+
+			return combinedMessage
+		},
+
+		toggleCollapsed(messages) {
+			this.$set(messages, 'collapsed', !messages.collapsed)
+		},
+
+		getNextMessageId(message) {
+			const nextMessage = this.messages[this.messages.findIndex(searchedMessage => searchedMessage.id === message.id) + 1]
+			return nextMessage?.id || this.nextMessageId
+		},
+
+		getPrevMessageId(message) {
+			const prevMessage = this.messages[this.messages.findIndex(searchedMessage => searchedMessage.id === message.id) - 1]
+			return prevMessage?.id || this.previousMessageId
+		},
+	},
+}
+</script>
+
+<style lang="scss" scoped>
+@import '../../../assets/variables';
+
+.message-group {
+	&__date-header {
+		display: block;
+		text-align: center;
+		padding-top: 20px;
+		position: relative;
+		margin: 20px 0;
+		.date {
+			margin-right: $clickable-area * 2;
+			content: attr(data-date);
+			padding: 4px 12px;
+			left: 50%;
+			color: var(--color-text-maxcontrast);
+			background-color: var(--color-background-dark);
+			border-radius: var(--border-radius-pill);
+		}
+	}
+
+	&__system {
+		display: flex;
+		flex-direction: column;
+	}
+}
+
+.wrapper {
+	max-width: $messages-list-max-width;
+	display: flex;
+	margin: auto;
+	padding: 0;
+	&--system {
+		flex-direction: column;
+		padding-left: $clickable-area + 8px;
+	}
+	&:focus {
+		background-color: rgba(47, 47, 47, 0.068);
+	}
+}
+
+.messages {
+	flex: auto;
+	display: flex;
+	flex-direction: column;
+	width: 100%;
+	min-width: 0;
+
+	&--header {
+		position: relative;
+		& &__toggle {
+			position: absolute;
+			top: 50%;
+			right: 0;
+			transform: translateY(-50%);
```

Comment:
```suggestion
			top: -6px;
			right: 0;
```

![image](https://github.com/nextcloud/spreed/assets/93392545/10d2390d-471e-49eb-b406-213310a96ab0)
_Note to self_